### PR TITLE
commands: a second API for `revaultd`

### PR DESCRIPTION
**Pull request by darosior**
This moves the implementation of the RPC commands from `jsonrpc/api/mod.rs` to a new `command` module, where it can then be exposed as a Rust interface for a downstream application (such as the GUI) to use in place of the JSONRPC-across-UDS which became clumsy since we made `revaultd` a library. This is a followup to https://github.com/revault/revaultd/pull/344, but does not remove the Windows code just yet.

An effort was put on documentation, and bikeshedding on the API is very welcome.
Unfortunately the PR is very convoluted but cleaning it up at this point would require a week.. Hopefully `--color-moved=zebra` can help at least a tiny bit, still.

It's still not final but i wanted to open this PR early to get feedback on the API.

### CODE REVIEW

**Comment by darosior** [comment]
For instance, we could have the commands be in an `impl DaemonHandler` to avoid its members to be public but isn't it overkill?.. Happy to have feedback on that or even suggestion of entirely different approaches.

**Comment by darosior** [comment]
I'm not so fond of the API, but let's call it ready for review.

File: src/jsonrpc/api/mod.rs [hunk 1]
```diff
@@ -304,56 +230,17 @@ impl RpcApi for RpcImpl {
         log::info!("Stopping revaultd");
 
         meta.rpc_utils.bitcoind_conn.shutdown();
-        meta.rpc_utils
-            .sigfetcher_tx
-            .send(SigFetcherMessageOut::Shutdown)
-            .map_err(Error::from)?;
+        meta.rpc_utils.sigfetcher_conn.shutdown();
```
[the remark this answers is not in the document]
Comment (darosior) [reply src/jsonrpc/api/mod.rs:233]: Right, i'm working on a modified version of the patch you gave me, which removes this.

**Comment by darosior** [comment]
Note to self: I also need to make `getdepositaddress` a command if i remove public access to `RevaultD`.

**Comment by darosior** [comment]
Alright, the API is much better now. Resulted from discussions with @edouardparis offline.

**Comment by darosior** [comment]
Sigh Windows...

**Comment by darosior** [comment]
Hopefully fixed Windows build. That's now ready for review.

**Review by edouardparis — APPROVED** [review]
ACK 948ac50abc54fdf44960626155a30a7b2ebf0629

**Pull request merged**